src/http.c

```diff
@@ -483,12 +508,53 @@ http_stream_setup(char **stream, const char *url)
   while ((line = evbuffer_readln(ctx.input_body, NULL, EVBUFFER_EOL_ANY)) && (n < 10))
     {
       n++;
-      if (strncasecmp(line, "http://", strlen("http://")) == 0)
+      if (is_m3u)
 	{
-	  DPRINTF(E_DBG, L_HTTP, "Found internet playlist stream (line %d): %s\n", n, line);
+          if ( strncasecmp(line, "http://", sizeof("http://")-1) == 0 )
+            {
+              DPRINTF(E_DBG, L_HTTP, "Found internet playlist (m3u) stream (line %d): %s\n", n, line);
 
-	  n = -1;
-	  break;
+              n = -1;
+              break;
+            }
+	}
+      else
+	{
+          if (!in_playlist)
+            {
+              if (strncasecmp(line, "[playlist]", sizeof("[playlist]")-1) == 0)
+                in_playlist = true;
+            }
+          else
+            {
+              pos = line;
+              while (*pos == ' ')
+                  ++pos;
+
+              if (strncasecmp(pos, "file", sizeof("file")-1) == 0)
+                {
+                  while (*pos != '=' && *pos != '\0')
+                      ++pos;
+                  if (*pos != '\0')
+                    {
+                      ++pos;
+                      while (*pos == ' ')
+                        ++pos;
+
+                      if (strncasecmp(pos, "http://", sizeof("http://")-1) == 0)
```

Comment:
Again better to continue on not found. So the general pattern (yes, there are plenty of exceptions!) is:
```
do_something
if (not found/failed/done)
  stop

do_something
if (not found/failed/done)
  stop

all done
```
instead of
```
do_something
if (ok)
  do_something
  if (ok)
    do_something
    if (ok)
       all done
```

Reply:
I just put out the first draft version that worked to see if you would want it, or if you had plans to unify the playlist parsing.